src/player.js

```diff
@@ -1239,15 +1256,46 @@ export default class Player extends FakeEventTarget {
    */
   _setDefaultTracks(): void {
     const activeTracks = this.getActiveTracks();
-    const playbackConfig = this._config.playback;
+    const playbackConfig = this.config.playback;
+    const offTextTrack: ?Track = this._getTracksByType(TrackTypes.TEXT).find(track => TextTrack.langComparer(OFF, track.language));
 
     this.hideTextTrack();
 
-    const textLanguage = (playbackConfig.textLanguage === "auto") ? Locale.language : playbackConfig.textLanguage;
-    this._setDefaultTrack(TrackTypes.TEXT, textLanguage, activeTracks.text);
+    this._setDefaultTrack(TrackTypes.TEXT, this._getLanguage(playbackConfig.textLanguage, activeTracks.text, TrackTypes.TEXT), offTextTrack);
     this._setDefaultTrack(TrackTypes.AUDIO, playbackConfig.audioLanguage, activeTracks.audio);
   }
 
+
+  /**
+   * Gets the text value that should be set in the captions - from the browser / activeTrack if the confiuuration is Auto
+   * @param {string} language - the language, in case the configuration is not auto
+   * @param {Track} activeTracks - active track data
+   * @param {string} type - the track type (text / audio)
+   * @private
+   * @returns {string}
+   */
+
+  _getLanguage(language: string, activeTrack: ?Track, type: string): string {
+
+
+    if (language === AUTO) {
+      const localLanguage = Locale.language;
+      const textTracks = this._getTracksByType(type);
+      const localeTrack: ?Track = textTracks.find(track => TextTrack.langComparer(localLanguage, track.language));
+
+      if (localeTrack) {
+        language = localeTrack.language;
+      } else if (activeTrack && Object.keys(activeTrack).length) {
```

Comment:
why the 
` Object.keys(activeTrack).length`
check?

Reply:
No, it can be undefined and therefore the first condition is enough.

Btw, to check empty object use Utils.Object.isEmptyObject and not like u did